Find a functionally equivalent code:android.app.SharedPreferencesImpl.EditorImpl.commitToMemory:COMMENT
Method Modifier: private     
Comment:// Returns true if any changes were made

@@ -27,40 +27,39 @@
             for (Map.Entry<String, Object> e : mModified.entrySet()) {
                 String k = e.getKey();
                 Object v = e.getValue();
-                if (v == this) {
-                    // magic value for a removal mutation
+                // equivalent to calling remove on that key.
+                if (v == this || v == null) {
                     if (!mMap.containsKey(k)) {
                         continue;
                     }
                     mMap.remove(k);
                 } else {
-                    boolean isSame = false;
                     if (mMap.containsKey(k)) {
                         Object existingValue = mMap.get(k);
                         if (existingValue != null && existingValue.equals(v)) {

